src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -131,14 +131,14 @@ public void setProvider(String clientName, FeatureProvider provider) {
     /**
      * Set the default provider and wait for initialization to finish.
      */
-    public void setProviderAndWait(FeatureProvider provider) {
+    public void setProviderAndWait(FeatureProvider provider) throws RuntimeException {
         try (AutoCloseableLock __ = lock.writeLockAutoCloseable()) {
             providerRepository.setProvider(
                     provider,
                     this::attachEventProvider,
                     this::emitReady,
                     this::detachEventProvider,
-                    this::emitError,
+                    this::emitErrorAndThrow,
```

Comment:
This is the main change - I pass a new delegate which not only emits the events, but throws.